libbeat/common/kubernetes/util.go

```diff
@@ -30,10 +30,25 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 
+	"github.com/pkg/errors"
+
 	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
-const defaultNode = "localhost"
+const namespaceFilePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+type discoveryUtils struct {
+	eDisc       HostDiscovery
+	client      kubernetes.Interface
+	isInCluster bool
+}
+type HostDiscovery interface {
```

Comment:
This interface might include a `GetNodeName()` function that will later be used in line 127